### PR TITLE
Fix the wrong translation of the coordinate in X11 platform

### LINKED ISSUE #75292: Windows move downward by the height of their title bars each time they popup (DisplayServer regression)
### Godot version

v4.0.1.stable.custom_build [cacf49999]

### System information

Fedora 37 (GNOME/X11)

### Issue description

Open and close the Editor Settings window repeatedly. Note how it moves downward each time it opens. I measured it, and it appears to move by exactly the height of the title bar each time. There is probably a window position calculation that does not account for the title bar height properly.

This issue does not happen with Single Window Mode enabled.

I have only tried this on GNOME with X11 as the windowing system.

### Steps to reproduce

See above

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by n0tank3sh**
Fix the wrong translation of the coordinate in DisplayServerX11::_window_changed.
Fix #75292

### CODE REVIEW

File: platform/linuxbsd/x11/display_server_x11.cpp [hunk 1]
```diff
@@ -29,6 +29,7 @@
 /**************************************************************************/
 
 #include "display_server_x11.h"
+#include <X11/Xlib.h>
```
Comment (bruvzg) [inline platform/linuxbsd/x11/display_server_x11.cpp:32]: Already included in the `display_server_x11.h`.

File: platform/linuxbsd/x11/display_server_x11.cpp [hunk 2]
```diff
@@ -3514,9 +3515,11 @@ void DisplayServerX11::_window_changed(XEvent *event) {
 		//the position in xconfigure is not useful here, obtain it manually
 		int x = 0, y = 0;
 		Window child;
+		XWindowAttributes xwa;
 		XTranslateCoordinates(x11_display, wd.x11_window, DefaultRootWindow(x11_display), 0, 0, &x, &y, &child);
-		new_rect.position.x = x;
-		new_rect.position.y = y;
+		XGetWindowAttributes(x11_display, wd.x11_window, &xwa);
+		new_rect.position.x = x - xwa.x;
+		new_rect.position.y = y - xwa.y;
```
Comment (bruvzg) [inline platform/linuxbsd/x11/display_server_x11.cpp:3522]: `position` is "client rect" origin not a window origin, it should not include decorations and will result in misplaced popups menus/other dropdowns. So this change seem to be wrong, and the issue should be somewhere else.

File: platform/linuxbsd/x11/display_server_x11.cpp (hunk 1 above)
Comment (MightiestGoat) [reply platform/linuxbsd/x11/display_server_x11.cpp:32]: My Youcompleteplugin did that. I didn't realize it. My bad.

**Comment by bruvzg** [comment]
Some extra info:

The issue seems to be random order of ChangeNofity, it is sent when window decoration is set and wine window is moved, order of events is random, and I do not see any way to tell them apart. Depending on the timing of events, `wd.position` might be set to a correct or incorrect value. So I'm not sure if it's possible to fix it at all.

**Comment by MightiestGoat** [comment]
@bruvzg During the processing of the ChangeNotify event, we are pulling the value for the position from XTranslateCoordinate which should give the accurate position of the window. Why there ever be a change in that?

**Comment by MightiestGoat** [comment]
I am trying to use gdb to debug this. I set a breakpoint to DisplayServerX11::window_set_position but it is not hitting it. Another thing I noticed is that in DisplayServerX11::window_set_position we are not changing the position of the window DisplayServerX11 side.

**Comment by bruvzg** [comment]
When showing a new window, ChangeNotify event is triggered 3–4 times with the different positions (some of these positions include decorations, others do not), I have no idea how to handle it.

**Comment by MightiestGoat** [comment]
Should we look into other projects on how they did it?

**Comment by MightiestGoat** [comment]
@bruvzg I snoop around for the projects. Then I decided to look at the documentation according to that. According to x.org
The X server can report ConfigureNotify events to clients wanting information about actual changes to a window's state, such as size, position, border, and stacking order. The X server generates this event type whenever one of the following configure window requests made by a client application actually completes:

A window's size, position, border, and/or stacking order is reconfigured by calling [XConfigureWindow](https://x.org/releases/current/doc/libX11/libX11/libX11.html#XConfigureWindow).

The window's position in the stacking order is changed by calling [XLowerWindow](https://x.org/releases/current/doc/libX11/libX11/libX11.html#XLowerWindow), [XRaiseWindow](https://x.org/releases/current/doc/libX11/libX11/libX11.html#XRaiseWindow), or [XRestackWindows](https://x.org/releases/current/doc/libX11/libX11/libX11.html#XRestackWindows).

A window is moved by calling [XMoveWindow](https://x.org/releases/current/doc/libX11/libX11/libX11.html#XMoveWindow).

A window's size is changed by calling [XResizeWindow](https://x.org/releases/current/doc/libX11/libX11/libX11.html#XResizeWindow).

A window's size and location is changed by calling [XMoveResizeWindow](https://x.org/releases/current/doc/libX11/libX11/libX11.html#XMoveResizeWindow).

A window is mapped and its position in the stacking order is changed by calling [XMapRaised](https://x.org/releases/current/doc/libX11/libX11/libX11.html#XMapRaised).

A window's border width is changed by calling [XSetWindowBorderWidth](https://x.org/releases/current/doc/libX11/libX11/libX11.html#XSetWindowBorderWidth).

So it is sent after the changes are made so acting upon shouldn't be the same as applying the same changes?

**Comment by MightiestGoat** [comment]
After hopelessly trying everything from changing the window to the override direct so the effect of the wm(which cannot be changed) and a few other things. I conclude I cannot solve this issue.

**Comment by MightiestGoat** [comment]
@bruvzg  those ChangeNotify events are being sent by the window manager.  There is another thing we cannot restore the window position by XMoveWindow even with _NET_FRAME_EXTENDS. The reason for that is that window managers are reparenting the window thus we are losing the initial position of the window. In conclusion, we need to find the parent of the window and save it somewhere even some decoration reparent the reparented window( parenting Inception right there).

**Comment by MightiestGoat** [comment]
Supersedes by #75709

**Pull request closed without merging**